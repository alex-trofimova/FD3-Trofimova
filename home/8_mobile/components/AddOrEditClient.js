import React from 'react';
import PropTypes from 'prop-types';

import {mobileCompanyEvents} from './events';

import './AddOrEditClient.css';

class AddOrEdiClient extends React.PureComponent {
    static propTypes = {
        editingOrNewClient: PropTypes.shape({
          id: PropTypes.number.isRequired,
          fam: PropTypes.string.isRequired,
          im: PropTypes.string.isRequired,
          otch: PropTypes.string.isRequired,
          balance: PropTypes.number.isRequired,
          status: PropTypes.string.isRequired,
        }),
        usedRegime: PropTypes.number.isRequired,
    };

    state = {
        editingOrNewClient: this.props.editingOrNewClient,
        usedRegime: this.props.usedRegime,
    };

    componentWillReceiveProps = (newProps) => {
        this.setState( {editingOrNewClient:newProps.editingOrNewClient, usedRegime:newProps.usedRegime} );
    };

    //использование ref для работы с полями формы
    idTextRef = null;
    famTextRef = null;
    imTextRef = null;
    otchTextRef = null;
    balanceTextRef = null;

    setIdTextRef = (ref) => {
        this.idTextRef=ref;
    };
    setFamTextRef = (ref) => {
        this.famTextRef=ref;
    };
    setImTextRef = (ref) => {
        this.imTextRef=ref;
    };
    setOtchTextRef = (ref) => {
        this.otchTextRef=ref;
    };
    setBalanceTextRef = (ref) => {
        this.balanceTextRef=ref;
    };
    
    //описание функции для добавления клиента по кнопке "Добавить"
    addNewClient = () => {
        let newAddingClient={};
        newAddingClient.id=this.state.editingOrNewClient.id;
        newAddingClient.fam=this.famTextRef.value;
        newAddingClient.im=this.imTextRef.value;
        newAddingClient.otch=this.otchTextRef.value;
        newAddingClient.balance=Number(this.balanceTextRef.value);

        if (newAddingClient.balance>0){
            newAddingClient.status='active';
        }
        else newAddingClient.status='blocked';

        mobileCompanyEvents.emit('EClientToAdd',newAddingClient);
    }

    cancelAction = () => {
        mobileCompanyEvents.emit('EClientToCancel');
    }

    //описание функции для сохранения изменений выбранного для редактирования клиента по кнопке "Сохранить изменения"
    saveEditedClient = () => {
        let EditedClient={};
        EditedClient.id=Number(this.state.editingOrNewClient.id)
        EditedClient.fam=this.famTextRef.value;
        EditedClient.im=this.imTextRef.value;
        EditedClient.otch=this.otchTextRef.value;
        EditedClient.balance=Number(this.balanceTextRef.value);
        if (EditedClient.balance>0){
            EditedClient.status='active';
        }
        else EditedClient.status='blocked';

        mobileCompanyEvents.emit('EClientToSaveChanges',EditedClient);
    }

    render() {

        console.log("AddOrEditClient id="+this.state.editingOrNewClient.id+" render");
        
        return (
            <div className="AddOrEditClient_card">
            <h3>
                { (this.state.usedRegime===1)
                  ? 'Редактировать выбранного клиента'
                  : 'Добавить нового клиента'
                }
              </h3>
            <div>
              <span className='AddOrEditClient_title'>Фамилия: </span>
              <input type='text' name='clientFam' className='AddOrEditClient_input'
                    ref={this.setFamTextRef}
                    defaultValue={this.state.editingOrNewClient.fam}
              />
            </div>
            <div>
              <span className='AddOrEditClient_title'>Имя: </span>
              <input type='text' name='clientIm' className='AddOrEditClient_input'
                    ref={this.setImTextRef}
                    defaultValue={this.state.editingOrNewClient.im}
              />
            </div>
            <div>
              <span className='AddOrEditClient_title'>Отчество: </span>
              <input type='text' name='clientOtch' className='AddOrEditClient_input'
                    ref={this.setOtchTextRef}
                    defaultValue={this.state.editingOrNewClient.otch}
              />
            </div>
            <div>
              <span className='AddOrEditClient_title'>Баланс: </span>
              <input type='text' name='clientBalance' className='AddOrEditClient_input'
                    ref={this.setBalanceTextRef}
                    defaultValue={this.state.editingOrNewClient.balance}
              />
            </div>
            
            { (this.state.usedRegime==1)
              ? <button className='AddOrEditClient_btn' value='add' onClick={this.saveEditedClient}>
                    Сохранить изменения
                  </button>
              : <button className='AddOrEditClient_btn' value='add' onClick={this.addNewClient}>
                    Добавить
                </button>
            }
            
            <button className='AddOrEditClient_btn' value='cancel' onClick={this.cancelAction}>Отмена</button>
          </div>
          
          )
      }
}

export default AddOrEdiClient;