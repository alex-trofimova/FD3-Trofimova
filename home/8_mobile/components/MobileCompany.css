.MobileCompany
{
  width: 800px;
  padding: 3px 7px;
}

.MobileCompanyName
{
  margin: 5px 0 10px 0;
}

.MobileCompanyClients
{
  padding-top: 5px;
}

.MobileCompanyClientsList {
  width: 100%;
  margin: 10px 0;
}

.TableHeader {
  border-bottom: 2px solid grey;
  text-align: left;
}


