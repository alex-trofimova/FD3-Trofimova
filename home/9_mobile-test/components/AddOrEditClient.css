.AddOrEditClient_title {
    display: inline-block;
    width: 100px;
  }
  
  .AddOrEditClient_btn{
    margin-top: 10px;
  }