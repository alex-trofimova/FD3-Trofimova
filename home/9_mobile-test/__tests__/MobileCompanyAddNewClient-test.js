import React from 'react';
import renderer from 'react-test-renderer';

import MobileCompany from '../components/MobileCompany';
import AddOrEditClient from '../components/AddOrEditClient';
let clientsArr=require('./json/clientsList.json');
let emptyclient=require('./json/emptyclient.json');
let newclient=require('./json/newclient.json');
let renewclientsArr=require('./json/renewclientsList.json');

let companyName='Velcom';

function createNodeMock(el) {
  if (el.type === 'input') {
    return {};
  }
  return null;
}

test('работа MobileCompany AddNewClient', () => {

// рендер компонента MobileCompany
const component1 = renderer.create(
  <MobileCompany 
  name={companyName}
  clients={clientsArr}
  />
);

// проверка утверждений
let component1Tree=component1.toJSON();
expect(component1Tree).toMatchSnapshot();

// находим в верстке компонента MobileCompany кнопку Добавить клиента 
const buttonElem1 = component1.root.find(  el => el.props.className == 'MobileCompanyNewClient_btn AddNewClientBtn' );
// и "нажмём" на неё
buttonElem1.props.onClick();

const options = {createNodeMock};
// обновление: рендер компонента AddOrEditClient
const component2 = renderer.create(
  <AddOrEditClient 
  editingOrNewClient={emptyclient}
  usedRegime={2}
  />, options
);

// проверка утверждений
component1Tree=component1.toJSON();
expect(component1Tree).toMatchSnapshot();
let component2Tree=component2.toJSON();
expect(component2Tree).toMatchSnapshot();

// обновление компонента AddOrEditClient
component2.update(<AddOrEditClient 
  editingOrNewClient={newclient}
  usedRegime={2}
  />);

// проверка утверждений
component1Tree=component1.toJSON();
expect(component1Tree).toMatchSnapshot();
component2Tree=component2.toJSON();
expect(component2Tree).toMatchSnapshot();

// находим в верстке компонента AddOrEditClient кнопку Добавить 
const buttonElem2 = component2.root.find(  el => el.props.className == 'AddOrEditClient_btn AddClientBtn' );
// и "нажмём" на неё
buttonElem2.props.onClick();

// обновление компонента MobileCompany
component1.update(<MobileCompany name={companyName}
clients={renewclientsArr}/>);

// удаление (отмонтирование) компонента AddOrEditClient
component2.unmount();

// проверка утверждений
component1Tree=component1.toJSON();
expect(component1Tree).toMatchSnapshot();

    
});
