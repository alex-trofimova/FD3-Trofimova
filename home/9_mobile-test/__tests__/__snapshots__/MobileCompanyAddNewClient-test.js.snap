// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа MobileCompany AddNewClient 1`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Velcom"
  />
  <input
    onClick={[Function]}
    type="button"
    value="MTS"
  />
  <div
    className="MobileCompanyName"
  >
    Компания: 
    Velcom
  </div>
  <div>
    <button
      className="MobileCompanyFilter_btn AllBtn"
      onClick={[Function]}
      value="all"
    >
      Все
    </button>
    <button
      className="MobileCompanyFilter_btn ActiveBtn"
      onClick={[Function]}
      value="active"
    >
      Активные
    </button>
    <button
      className="MobileCompanyFilter_btn BlockedBtn"
      onClick={[Function]}
      value="blocked"
    >
      Заблокированные
    </button>
  </div>
  <table
    className="MobileCompanyClientsList"
  >
    <thead
      className="HeaderContent"
    >
      <tr
        className="TableHeader"
      >
        <th
          className="HeaderCell"
        >
          Фамилия
        </th>
        <th
          className="HeaderCell"
        >
          Имя
        </th>
        <th
          className="HeaderCell"
        >
          Отчество
        </th>
        <th
          className="HeaderCell"
        >
          Баланс
        </th>
        <th
          className="HeaderCell"
        >
          Статус
        </th>
        <th
          className="HeaderCell"
        >
          Редактировать
        </th>
        <th
          className="HeaderCell"
        >
          Удалить
        </th>
      </tr>
    </thead>
    <tbody
      className="ClientsListContent"
    >
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Иванов
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Иван
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Иванович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          200
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Сидоров
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Сидор
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Сидорович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          250
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Петров
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Пётр
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Петрович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          180
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Григорьев
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Григорий
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Григорьевич
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          -220
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          blocked
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <button
    className="MobileCompanyNewClient_btn AddNewClientBtn"
    onClick={[Function]}
    value="add"
  >
    Добавить клиента
  </button>
</div>
`;

exports[`работа MobileCompany AddNewClient 2`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Velcom"
  />
  <input
    onClick={[Function]}
    type="button"
    value="MTS"
  />
  <div
    className="MobileCompanyName"
  >
    Компания: 
    Velcom
  </div>
  <div>
    <button
      className="MobileCompanyFilter_btn AllBtn"
      onClick={[Function]}
      value="all"
    >
      Все
    </button>
    <button
      className="MobileCompanyFilter_btn ActiveBtn"
      onClick={[Function]}
      value="active"
    >
      Активные
    </button>
    <button
      className="MobileCompanyFilter_btn BlockedBtn"
      onClick={[Function]}
      value="blocked"
    >
      Заблокированные
    </button>
  </div>
  <table
    className="MobileCompanyClientsList"
  >
    <thead
      className="HeaderContent"
    >
      <tr
        className="TableHeader"
      >
        <th
          className="HeaderCell"
        >
          Фамилия
        </th>
        <th
          className="HeaderCell"
        >
          Имя
        </th>
        <th
          className="HeaderCell"
        >
          Отчество
        </th>
        <th
          className="HeaderCell"
        >
          Баланс
        </th>
        <th
          className="HeaderCell"
        >
          Статус
        </th>
        <th
          className="HeaderCell"
        >
          Редактировать
        </th>
        <th
          className="HeaderCell"
        >
          Удалить
        </th>
      </tr>
    </thead>
    <tbody
      className="ClientsListContent"
    >
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Иванов
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Иван
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Иванович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          200
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Сидоров
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Сидор
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Сидорович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          250
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Петров
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Пётр
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Петрович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          180
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Григорьев
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Григорий
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Григорьевич
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          -220
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          blocked
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="AddOrEditClient_card"
  >
    <h3>
      Добавить нового клиента
    </h3>
    <div>
      <span
        className="AddOrEditClient_title"
      >
        Фамилия: 
      </span>
      <input
        className="AddOrEditClient_input"
        defaultValue=""
        name="clientFam"
        type="text"
      />
    </div>
    <div>
      <span
        className="AddOrEditClient_title"
      >
        Имя: 
      </span>
      <input
        className="AddOrEditClient_input"
        defaultValue=""
        name="clientIm"
        type="text"
      />
    </div>
    <div>
      <span
        className="AddOrEditClient_title"
      >
        Отчество: 
      </span>
      <input
        className="AddOrEditClient_input"
        defaultValue=""
        name="clientOtch"
        type="text"
      />
    </div>
    <div>
      <span
        className="AddOrEditClient_title"
      >
        Баланс: 
      </span>
      <input
        className="AddOrEditClient_input"
        defaultValue={0}
        name="clientBalance"
        type="text"
      />
    </div>
    <button
      className="AddOrEditClient_btn AddClientBtn"
      onClick={[Function]}
      value="add"
    >
      Добавить
    </button>
    <button
      className="AddOrEditClient_btn CancelBtn"
      onClick={[Function]}
      value="cancel"
    >
      Отмена
    </button>
  </div>
</div>
`;

exports[`работа MobileCompany AddNewClient 3`] = `
<div
  className="AddOrEditClient_card"
>
  <h3>
    Добавить нового клиента
  </h3>
  <div>
    <span
      className="AddOrEditClient_title"
    >
      Фамилия: 
    </span>
    <input
      className="AddOrEditClient_input"
      defaultValue=""
      name="clientFam"
      type="text"
    />
  </div>
  <div>
    <span
      className="AddOrEditClient_title"
    >
      Имя: 
    </span>
    <input
      className="AddOrEditClient_input"
      defaultValue=""
      name="clientIm"
      type="text"
    />
  </div>
  <div>
    <span
      className="AddOrEditClient_title"
    >
      Отчество: 
    </span>
    <input
      className="AddOrEditClient_input"
      defaultValue=""
      name="clientOtch"
      type="text"
    />
  </div>
  <div>
    <span
      className="AddOrEditClient_title"
    >
      Баланс: 
    </span>
    <input
      className="AddOrEditClient_input"
      defaultValue={0}
      name="clientBalance"
      type="text"
    />
  </div>
  <button
    className="AddOrEditClient_btn AddClientBtn"
    onClick={[Function]}
    value="add"
  >
    Добавить
  </button>
  <button
    className="AddOrEditClient_btn CancelBtn"
    onClick={[Function]}
    value="cancel"
  >
    Отмена
  </button>
</div>
`;

exports[`работа MobileCompany AddNewClient 4`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Velcom"
  />
  <input
    onClick={[Function]}
    type="button"
    value="MTS"
  />
  <div
    className="MobileCompanyName"
  >
    Компания: 
    Velcom
  </div>
  <div>
    <button
      className="MobileCompanyFilter_btn AllBtn"
      onClick={[Function]}
      value="all"
    >
      Все
    </button>
    <button
      className="MobileCompanyFilter_btn ActiveBtn"
      onClick={[Function]}
      value="active"
    >
      Активные
    </button>
    <button
      className="MobileCompanyFilter_btn BlockedBtn"
      onClick={[Function]}
      value="blocked"
    >
      Заблокированные
    </button>
  </div>
  <table
    className="MobileCompanyClientsList"
  >
    <thead
      className="HeaderContent"
    >
      <tr
        className="TableHeader"
      >
        <th
          className="HeaderCell"
        >
          Фамилия
        </th>
        <th
          className="HeaderCell"
        >
          Имя
        </th>
        <th
          className="HeaderCell"
        >
          Отчество
        </th>
        <th
          className="HeaderCell"
        >
          Баланс
        </th>
        <th
          className="HeaderCell"
        >
          Статус
        </th>
        <th
          className="HeaderCell"
        >
          Редактировать
        </th>
        <th
          className="HeaderCell"
        >
          Удалить
        </th>
      </tr>
    </thead>
    <tbody
      className="ClientsListContent"
    >
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Иванов
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Иван
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Иванович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          200
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Сидоров
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Сидор
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Сидорович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          250
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Петров
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Пётр
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Петрович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          180
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Григорьев
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Григорий
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Григорьевич
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          -220
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          blocked
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="AddOrEditClient_card"
  >
    <h3>
      Добавить нового клиента
    </h3>
    <div>
      <span
        className="AddOrEditClient_title"
      >
        Фамилия: 
      </span>
      <input
        className="AddOrEditClient_input"
        defaultValue=""
        name="clientFam"
        type="text"
      />
    </div>
    <div>
      <span
        className="AddOrEditClient_title"
      >
        Имя: 
      </span>
      <input
        className="AddOrEditClient_input"
        defaultValue=""
        name="clientIm"
        type="text"
      />
    </div>
    <div>
      <span
        className="AddOrEditClient_title"
      >
        Отчество: 
      </span>
      <input
        className="AddOrEditClient_input"
        defaultValue=""
        name="clientOtch"
        type="text"
      />
    </div>
    <div>
      <span
        className="AddOrEditClient_title"
      >
        Баланс: 
      </span>
      <input
        className="AddOrEditClient_input"
        defaultValue={0}
        name="clientBalance"
        type="text"
      />
    </div>
    <button
      className="AddOrEditClient_btn AddClientBtn"
      onClick={[Function]}
      value="add"
    >
      Добавить
    </button>
    <button
      className="AddOrEditClient_btn CancelBtn"
      onClick={[Function]}
      value="cancel"
    >
      Отмена
    </button>
  </div>
</div>
`;

exports[`работа MobileCompany AddNewClient 5`] = `
<div
  className="AddOrEditClient_card"
>
  <h3>
    Добавить нового клиента
  </h3>
  <div>
    <span
      className="AddOrEditClient_title"
    >
      Фамилия: 
    </span>
    <input
      className="AddOrEditClient_input"
      defaultValue="Trofimova"
      name="clientFam"
      type="text"
    />
  </div>
  <div>
    <span
      className="AddOrEditClient_title"
    >
      Имя: 
    </span>
    <input
      className="AddOrEditClient_input"
      defaultValue="Alexandra"
      name="clientIm"
      type="text"
    />
  </div>
  <div>
    <span
      className="AddOrEditClient_title"
    >
      Отчество: 
    </span>
    <input
      className="AddOrEditClient_input"
      defaultValue="Vladimirovna"
      name="clientOtch"
      type="text"
    />
  </div>
  <div>
    <span
      className="AddOrEditClient_title"
    >
      Баланс: 
    </span>
    <input
      className="AddOrEditClient_input"
      defaultValue={400}
      name="clientBalance"
      type="text"
    />
  </div>
  <button
    className="AddOrEditClient_btn AddClientBtn"
    onClick={[Function]}
    value="add"
  >
    Добавить
  </button>
  <button
    className="AddOrEditClient_btn CancelBtn"
    onClick={[Function]}
    value="cancel"
  >
    Отмена
  </button>
</div>
`;

exports[`работа MobileCompany AddNewClient 6`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Velcom"
  />
  <input
    onClick={[Function]}
    type="button"
    value="MTS"
  />
  <div
    className="MobileCompanyName"
  >
    Компания: 
    Velcom
  </div>
  <div>
    <button
      className="MobileCompanyFilter_btn AllBtn"
      onClick={[Function]}
      value="all"
    >
      Все
    </button>
    <button
      className="MobileCompanyFilter_btn ActiveBtn"
      onClick={[Function]}
      value="active"
    >
      Активные
    </button>
    <button
      className="MobileCompanyFilter_btn BlockedBtn"
      onClick={[Function]}
      value="blocked"
    >
      Заблокированные
    </button>
  </div>
  <table
    className="MobileCompanyClientsList"
  >
    <thead
      className="HeaderContent"
    >
      <tr
        className="TableHeader"
      >
        <th
          className="HeaderCell"
        >
          Фамилия
        </th>
        <th
          className="HeaderCell"
        >
          Имя
        </th>
        <th
          className="HeaderCell"
        >
          Отчество
        </th>
        <th
          className="HeaderCell"
        >
          Баланс
        </th>
        <th
          className="HeaderCell"
        >
          Статус
        </th>
        <th
          className="HeaderCell"
        >
          Редактировать
        </th>
        <th
          className="HeaderCell"
        >
          Удалить
        </th>
      </tr>
    </thead>
    <tbody
      className="ClientsListContent"
    >
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Иванов
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Иван
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Иванович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          200
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Сидоров
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Сидор
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Сидорович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          250
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Петров
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Пётр
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Петрович
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          180
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          active
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        >
          Григорьев
        </td>
        <td
          className="ClientCell ClientIm"
        >
          Григорий
        </td>
        <td
          className="ClientCell ClientOtch"
        >
          Григорьевич
        </td>
        <td
          className="ClientCell ClientBalance"
        >
          -220
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          blocked
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="ClientCell ClientFam"
        />
        <td
          className="ClientCell ClientIm"
        />
        <td
          className="ClientCell ClientOtch"
        />
        <td
          className="ClientCell ClientBalance"
        >
          NaN
        </td>
        <td
          className="ClientCell ClientStatus"
        >
          blocked
        </td>
        <td
          className="ClientCell ClientEdit"
        >
          <button
            className="ClientBtn EditBtn"
            onClick={[Function]}
          >
            редактировать
          </button>
        </td>
        <td
          className="ClientCell ClientDelete"
        >
          <button
            className="ClientBtn DeleteBtn"
            onClick={[Function]}
          >
            удалить
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <button
    className="MobileCompanyNewClient_btn AddNewClientBtn"
    onClick={[Function]}
    value="add"
  >
    Добавить клиента
  </button>
</div>
`;
